src/frontend/store/data-context.js

```diff
@@ -0,0 +1,44 @@
+
+/**
+ * WordPress dependencies
+ */
+import { createContext, useContext, useState, useEffect } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+import { format } from '@wordpress/date';
+
+const EventDataContext = createContext( [] );
+
+export function useEventData() {
+	const context = useContext( EventDataContext );
+	if ( context === undefined ) {
+		throw new Error( 'useEvents must be used within a Provider' );
+	}
+	return context;
+}
+
+export function EventDataProvider( { children, data } ) {
+	const [ eventData, setEventData ] = useState( [] );
+	// we bail out early if data is hydrated from the server.
+	if ( data ) {
```

Comment:
I think we should keep this in a ref to save the reference between retenders, but I don’t think we will need it unless we move this logic to `EventProvider`